yb-voyager/src/metadb/metadataDB.go

```diff
@@ -460,6 +460,32 @@ func (m *MetaDB) GetSegmentsToBeDeleted() ([]utils.Segment, error) {
 	return segmentsToBeDeleted, nil
 }
 
+func (m *MetaDB) MaxUnarchivedSegmentNum() (int, error) {
+	// sample query: SELECT MAX(segment_no) FROM queue_segment_meta WHERE archived = 0;
+	predicate := "archived = 0"
+	segments, err := m.querySegments(predicate)
+	if err != nil {
+		return -1, fmt.Errorf("fetch max unarchived segment num: %v", err)
+	}
+	if len(segments) == 0 {
+		return -1, nil
+	}
+	return segments[len(segments)-1].Num, nil
+}
+
+func (m *MetaDB) GetOngoingSegmentNum() (int, error) {
+	// sample query: SELECT MAX(segment_no) FROM queue_segment_meta WHERE imported_by_target_db_importer + imported_by_ff_db_importer = 0;
+	predicate := "imported_by_target_db_importer + imported_by_ff_db_importer + imported_by_fb_db_importer = 0"
```

Comment:
Shouldn't the predicate be:
`imported_by_target_db_importer + imported_by_ff_db_importer + imported_by_fb_db_importer != importer_count`